### ISSUE TITLE
package `fake-factory` is being deprecated in favour of `faker`

### ISSUE THREAD

**Issue by danielknell** [HypothesisWorks/hypothesis-python#394]
>  The fake-factory package will be deprecated on December 15th, 2016.

https://pypi.python.org/pypi/fake-factory

this might effect people doing:

```python
pip install hypothesis[fake-factory]
```

**Comment by alexwlchan** [comment]
Those docs don't say what deprecated means. I hope it just means "nasty warning if you run `pip install fake-factory`", not "running `pip install fake-factory` will fail to work".

I might have a bit of time to look at moving from fake-factory to faker in the next few weeks; might miss the deprecation deadline. Depends how much has changed in the meantime, and how many tests break when I try.

**Comment by danielknell** [comment]
aye, it is less than helpful...

**Comment by alexwlchan** [comment]
Ah, looks like this was started in #362, and the only test failures seem to be that we try to `pip uninstall fake-factory`, which unsurprisingly fails. I'll probably just build a small delta atop that PR.

**Comment by alexwlchan** [comment]
This is partially addressed by #395, but if you do `pip install hypothesis` you’ll still get the fake-factory. We should probably cut a new release at some point before fake-factory gets deprecated, and I should have written a changelog entry for it. Oops.

**Comment by Daenyth** [comment]
Will there be any large changes to know about? I use fake_factory in some tests right now.

**Comment by DRMacIver** [comment]
I don't think there are any changes other than that you might have to update your requirements.txt to point to the new package name if you depend on it explicitly.

**Comment by skorokithakis** [comment]
The whole thing fails to work now:

~~~
  File "main/tests.py", line 4, in <module>
    from hypothesis.extra.django.models import models, default_value
  File "lib/python3.5/site-packages/hypothesis/extra/django/models.py", line 27, in <module>
    import hypothesis.extra.fakefactory as ff
  File "lib/python3.5/site-packages/hypothesis/extra/fakefactory.py", line 23, in <module>
    import faker
  File "lib/python3.5/site-packages/faker/__init__.py", line 7, in <module>
    raise ImportError(error)
ImportError: The ``fake-factory`` package is now called ``Faker``.

Please update your requirements.
~~~

**Comment by DRMacIver** [comment]
I think that's because you need to update your requirements, and doesn't really have anything to do with Hypothesis.

If you install through hypothesis[fakefactory] you get Faker. If you installed fake-factory directly you'll need to update your requiremnets to change that.

**Comment by alexwlchan** [comment]
@DRMacIver Don’t think so – looks like the new version of `setup.py` that updates the dependency from faker to fake-factory is only in Git, not on PyPI. If I install Hypothesis in a fresh virtualenv:

```console
$ pip install hypothesis[fakefactory]
Collecting hypothesis[fakefactory]
Collecting enum34 (from hypothesis[fakefactory])
  Using cached enum34-1.1.6-py2-none-any.whl
Collecting fake-factory>=0.6.0 (from hypothesis[fakefactory])
  Downloading fake_factory-9999.9.9-py2.py3-none-any.whl (556kB)
    100% |████████████████████████████████| 557kB 605kB/s
Installing collected packages: enum34, fake-factory, hypothesis
Successfully installed enum34-1.1.6 fake-factory-9999.9.9 hypothesis-3.6.0

$ pip freeze
enum34==1.1.6
fake-factory==9999.9.9
hypothesis==3.6.0
wheel==0.24.0

$ python -c 'import hypothesis.extra.fakefactory'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/alexwlchan/.virtualenvs/tempenv-3411163925956/lib/python2.7/site-packages/hypothesis/extra/fakefactory.py", line 23, in <module>
    import faker
  File "/Users/alexwlchan/.virtualenvs/tempenv-3411163925956/lib/python2.7/site-packages/faker/__init__.py", line 7, in <module>
    raise ImportError(error)
ImportError: The ``fake-factory`` package is now called ``Faker``.

Please update your requirements.
```

**Comment by skorokithakis** [comment]
Yeah, I didn't install `fake-factory` manually. I assume hypothesis requires anything over 0.7.2, which resolves to the deprecated package. Either use faker or pin to 0.7.2 (at least for now).

**Comment by DRMacIver** [comment]
Ah, right, yes. My bad, sorry, I forgot we hadn't released a version with that change.

I'll do a release of Hypothesis in the next day or two. In the meantime, I recommend just adding faker to your requirements.txt and removing hypothesis[fakefactory' from it.

**Comment by skorokithakis** [comment]
I only have `hypothesis[django]` in it, but what's worked is just changing the requirement to `fake-factory==0.7.2` for now (although I use `pip-compile`, so everything is pinned exactly).

**Comment by alexwlchan** [comment]
This should be fixed in Hypothesis 3.6.1, which uses faker instead of fake-factory.
